src/sinks/datadog_metrics.rs

```diff
@@ -0,0 +1,377 @@
+#[derive(Debug, Clone, PartialEq, Serialize)]
+struct DatadogRequest {
+    series: Vec<DatadogMetric>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
+struct DatadogMetric {
+    metric: String,
+    r#type: DatadogMetricType,
+    interval: Option<i64>,
+    points: Vec<DatadogPoint>,
+    tags: Option<Vec<String>>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum DatadogMetricType {
+    Gauge,
+    Count,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
+struct DatadogPoint(i64, f64);
+
+#[typetag::serde(name = "datadog")]
+impl SinkConfig for DatadogConfig {
+    fn build(&self, acker: Acker) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let sink = DatadogSvc::new(self.clone(), acker)?;
+        let healthcheck = DatadogSvc::healthcheck(self.clone())?;
+        Ok((sink, healthcheck))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Metric
+    }
+}
+
+impl DatadogSvc {
+    pub fn new(config: DatadogConfig, acker: Acker) -> crate::Result<super::RouterSink> {
+        let batch_size = config.batch_size.unwrap_or(20);
+        let batch_timeout = config.batch_timeout.unwrap_or(1);
+
+        let timeout = config.request_timeout_secs.unwrap_or(60);
+        let in_flight_limit = config.request_in_flight_limit.unwrap_or(5);
+        let rate_limit_duration = config.request_rate_limit_duration_secs.unwrap_or(1);
+        let rate_limit_num = config.request_rate_limit_num.unwrap_or(5);
+        let retry_attempts = config.request_retry_attempts.unwrap_or(5);
+        let retry_backoff_secs = config.request_retry_backoff_secs.unwrap_or(1);
+
+        let policy = FixedRetryPolicy::new(
+            retry_attempts,
+            Duration::from_secs(retry_backoff_secs),
+            HttpRetryLogic,
+        );
+
+        let uri = format!("{}/api/v1/series?api_key={}", config.host, config.api_key)
+            .parse::<Uri>()
+            .context(super::UriParseError)?;
+
+        let http_service = HttpService::new(move |body: Vec<u8>| {
+            let mut builder = hyper::Request::builder();
+            builder.method(Method::POST);
+            builder.uri(uri.clone());
+
+            builder.header("Content-Type", "application/json");
+            builder.body(body).unwrap()
+        })?;
+
+        let datadog_http_service = DatadogSvc {
+            config,
+            state: DatadogState {
+                last_sent_timestamp: Utc::now().timestamp(),
+            },
+            inner: http_service,
+        };
+
+        let service = ServiceBuilder::new()
+            .concurrency_limit(in_flight_limit)
+            .rate_limit(rate_limit_num, Duration::from_secs(rate_limit_duration))
+            .retry(policy)
+            .timeout(Duration::from_secs(timeout))
+            .service(datadog_http_service);
+
+        let sink = BatchServiceSink::new(service, acker).batched_with_min(
```

Comment:
Do we need this to be a minimum batch size? It seems like we should be fine sending whatever size batch we have available without waiting around for more.

Reply:
Nope, we don't actually need it here.

Do you think it makes sense to introduce something like `batched_with_delay` that will require only `batch_timeout`config item?

Reply:
We could, but I'm not sure that we need it. Why do you want to force a delay in the batching?

The idea of not having something like that is to let the batching happen "naturally" as throughput increases. Sinks should only support a handful of in-flight requests at a time, so a high throughput situation would hit that limit and cause batching to happen all by itself, without us having to set a specific delay. If you're not hitting that limit, the idea is that you probably don't need the batching anyway.